firmware/src/sensors/sensors_task.cpp

```diff
@@ -239,24 +286,210 @@ void SensorsTask::run()
             sensors_state.illumination.lux_adj = luminosity_adjustment;
 
-            if (millis() % 1000 == 0)
-            {
-                snprintf(buf_, sizeof(buf_), "Illumination sensor: millilux: %.2f, avg %.2f, adj %.2f", lux * 1000, lux_avg * 1000, luminosity_adjustment);
-                LOGV(PB_LogLevel_DEBUG, buf_);
-            }
             last_illumination_check_ms = millis();
         }
 #endif
+
+        if (millis() - log_ms > 1000)
+        {
+            LOGV(PB_LogLevel_DEBUG, "System temp %0.2f °C", last_system_temperature);
+            LOGV(PB_LogLevel_DEBUG, "Proximity sensor:  range %d, distance %dmm", measure.RangeStatus, measure.RangeMilliMeter);
+#if SK_STRAIN
+            LOGV(PB_LogLevel_DEBUG, "Strain: reading:\n        Virtual button code: %d\n        Strain value: %f\n        Press value: %f", sensors_state.strain.virtual_button_code, sensors_state.strain.raw_value, press_value_unit);
+#endif
+#if SK_ALS
+            LOGV(PB_LogLevel_DEBUG, "Illumination sensor: millilux: %.2f, avg %.2f, adj %.2f", lux * 1000, lux_avg * 1000, luminosity_adjustment);
+#endif
+            log_ms = millis();
+        }
         delay(1);
     }
 }
 
+#if SK_STRAIN
+void SensorsTask::strainCalibrationCallback()
+{
+    PB_PersistentConfiguration config = configuration_->get();
+
+    if (config.strain_scale == 0 && strain.get_scale() == 1.0f)
+    {
+        LOGI("Strain sensor needs Factory Calibration, press 'Y' to begin!");
+        return;
+    }
+
+    // LOGD("CONFIG VALUES: %f %f", strain_calibration.idle_value, strain_calibration.press_delta);
+
+    if (strain_calibration_step_ == 0)
+    {
+        strain_calibration_step_ = 1;
+
+        LOGI("Strain calibration step 1: Don't touch the knob, then press 'S' again");
+        strain.set_offset(0);
+        strain.tare();
+        delay(200);
+    }
+    else if (strain_calibration_step_ == 1)
+    {
+        strain_calibration_step_ = 2;
+        LOGI("Strain calibration step 2: When desired weight is reached press 'S' again");
+    }
+    else if (strain_calibration_step_ == 2)
+    {
+        config.strain.press_delta = sensors_state.strain.raw_value;
+        config.has_strain = true;
+
+        LOGD("  press_delta=%f", config.strain.press_delta);
+        LOGD("  raw_value=%f", sensors_state.strain.raw_value);
+        LOGI("Saving strain calibration...");
+
+        updateStrainCalibration(config.strain.idle_value, sensors_state.strain.raw_value);
+
+        if (configuration_->setStrainCalibrationAndSave(config.strain))
+        {
+            LOGI("Strain calibration saved!");
+        }
+        else
+        {
+            LOGE("Strain calibration failed to save!");
+        }
+
+        delay(2000);
+
+        strain.set_offset(0);
+        strain.tare();
+        delay(200);
+
+        LOGI("Strain calibration complete!");
+
+        strain_calibration_step_ = 0;
+    }
+}
+
+void SensorsTask::factoryStrainCalibrationCallback()
```

Comment:
Could you write documentation for this function explaining logic of this calibration? So it would be a bit easier to understand.